crates/accelerate/src/commutation_checker.rs

```diff
@@ -0,0 +1,725 @@
+        out_dict.set_item("_cache_hit", self._cache_hit)?;
+        out_dict.set_item("cache", self.cache.clone())?;
+        out_dict.set_item("library", self.library.library.to_object(py))?;
+        out_dict.set_item("gates", self.gates.clone())?;
+        Ok(out_dict.unbind())
+    }
+
+    fn __setstate__(&mut self, py: Python, state: PyObject) -> PyResult<()> {
+        let dict_state = state.downcast_bound::<PyDict>(py)?;
+        self.cache_max_entries = dict_state
+            .get_item("cache_max_entries")?
+            .unwrap()
+            .extract()?;
+        self.current_cache_entries = dict_state
+            .get_item("current_cache_entries")?
+            .unwrap()
+            .extract()?;
+        self._cache_miss = dict_state.get_item("_cache_miss")?.unwrap().extract()?;
+        self._cache_hit = dict_state.get_item("_cache_hit")?.unwrap().extract()?;
+        self.library = CommutationLibrary {
+            library: dict_state.get_item("library")?.unwrap().extract()?,
+        };
+        self.cache = dict_state.get_item("cache")?.unwrap().extract()?;
+        self.gates = dict_state.get_item("gates")?.unwrap().extract()?;
+        Ok(())
+    }
+}
+
+impl CommutationChecker {
+    #[allow(clippy::too_many_arguments)]
+    fn commute_inner(
+        &mut self,
+        py: Python,
+        instr1: &CircuitInstruction,
+        qargs1: &[u32],
+        cargs1: &[u32],
+        instr2: &CircuitInstruction,
+        qargs2: &[u32],
+        cargs2: &[u32],
+        max_num_qubits: u32,
+    ) -> bool {
+        if let Some(gates) = &self.gates {
+            if !gates.is_empty()
+                && (!gates.contains(instr1.op().name()) || !gates.contains(instr2.op().name()))
+            {
+                return false;
+            }
+        }
+
+        let commutation: Option<bool> = commutation_precheck(
+            instr1,
+            qargs1,
+            cargs1,
+            instr2,
+            qargs2,
+            cargs2,
+            max_num_qubits,
+        );
+        if let Some(is_commuting) = commutation {
+            return is_commuting;
+        }
+        let op1 = instr1.op();
+        let op2 = instr2.op();
+        let reversed = if op1.num_qubits() != op2.num_qubits() {
+            op1.num_qubits() > op2.num_qubits()
+        } else {
+            BigInt::from_signed_bytes_be(op1.name().as_bytes())
+                >= BigInt::from_signed_bytes_be(op2.name().as_bytes())
+        };
+        let (first_instr, second_instr) = if reversed {
+            (instr2, instr1)
+        } else {
+            (instr1, instr2)
+        };
+        let (first_op, second_op) = if reversed { (op2, op1) } else { (op1, op2) };
+        let (first_qargs, second_qargs) = if reversed {
+            (qargs2, qargs1)
+        } else {
+            (qargs1, qargs2)
+        };
+        let (_first_cargs, _second_cargs) = if reversed {
+            (cargs2, cargs1)
+        } else {
+            (cargs1, cargs2)
+        };
```

Comment:
This block doesn't seem to be used by anything at the moment.
```suggestion
```

Reply:
Removed in 454208cd9